edgedb/ai/core.py

```diff
@@ -112,23 +125,26 @@ def query_rag(
     def stream_rag(
         self, message: str, context: typing.Optional[types.QueryContext] = None
     ):
-        if context is None:
-            context = self.context
         with httpx_sse.connect_sse(
             self.client,
             "post",
-            **types.RAGRequest(
-                model=self.options.model,
-                prompt=self.options.prompt,
+            **self._make_rag_request(
                 context=context,
-                query=message,
+                message=message,
                 stream=True,
             ).to_httpx_request(),
         ) as event_source:
             event_source.response.raise_for_status()
             for sse in event_source.iter_sse():
                 yield sse.data
 
+    def generate_custom_embeddings(self, *inputs: str, model: str):
```

Comment:
working on the JS version of this but my instinct was to name this `generateEmbeddings`. Is there a significance to the "custom" here? Does it help to denote that these are not the automatically indexed ones?

Reply:
Yeah, that was my intention, but I'm not sure this makes sense (like in English).

Reply:
I wonder if `custom` communicates enough to pay for the cost. `search` would be even more specific and point at the intended use case, but maybe that's too restrictive since you might use it outside of `ext::ai::search`? Does `generate_embeddings` make it seem too much like you're triggering something rather than doing this one-off embedding generation?

The only reason I ask is that having "custom" here would make me as a developer want to know more about what "custom" means and what other non-custom methods their might be. Maybe that's a good thing and worth adding here, but it feels a little like an unnecessary mental speed bump.

Reply:
> Does `generate_embeddings` make it seem too much like you're triggering something rather than doing this one-off embedding generation?

Yeah, I had the same struggle. I thought about `retrieve_embeddings()`, which is just worse.

> Maybe that's a good thing and worth adding here, but it feels a little like an unnecessary mental speed bump.

Right, I'll also change it to `generate_embeddings()` and add an explanation in the docs.